awspricing/offers.py

```diff
@@ -90,18 +95,38 @@ def _generate_reverse_sku_mapping(self,
                                       ):
         # type: (...) -> Dict[str, str]
 
+        """Generate a reverse mapping from a hash of product attributes to SKU.
+
+        Only hashes that are unique across all products will be included in the
+        result; products which collide on hash will be discarded.
+        """
+
         product_family = kwargs.get('product_family')
 
         result = {}  # type: Dict[str, str]
+
+        # There are cases where the set of attributes are not unique across all
+        # products and cause hash collisions. In these cases, we take note of
+        # the collision and do not include these products in the mapping.
+        attribute_collisions = set()
+
         for sku, product in six.iteritems(self._offer_data['products']):
             if (product_family is not None and
                     product['productFamily'] != product_family):
                 continue
             attrs = [product['attributes'][attr] for attr in attribute_names]
             key = self.hash_attributes(*attrs)
             if key in result:
-                raise ValueError("Attribute collision for: {}".format(key))
-            result[key] = sku
+                # There is an attribute collision, so do not include any of
```

Comment:
Yep, also not a huge fan of this but I think it's better not to break everything so easily and figure out some alternative approach to this later.